wallet/invoices.c

```diff
@@ -143,7 +220,11 @@ const struct invoice *invoices_create(struct invoices *invoices,
 	}
 
 	/* Compute expiration. */
-	expiry_time = time_now().ts.tv_sec + expiry;
+	expiry_time = now + expiry;
+	if (expiry == 0)
+		state = EXPIRED;
+	else
+		state = UNPAID;
```

Comment:
I would *always* start them as UNPAID, and let the timer take care of immediately expiring them through the normal path.

Less code == less bugs!

Reply:
Okay!